cupy/core/_routines_math.pyx

```diff
@@ -192,15 +342,20 @@ def _add_scan_blocked_sum_kernel(dtype):
         unsigned int idxAdd = idxBase - 1;
 
         if(idxAdded < n){
-            src_dst[idxAdded] += src_dst[idxAdd];
+            src_dst[idxAdded] ${op}= src_dst[idxAdd];
         }
     }
-    """).substitute(name=name, dtype=dtype)
+    """).substitute(name=name, dtype=dtype, op=ops[op])
     module = compile_with_cache(source)
     return module.get_function(name)
 
 
-cdef ndarray scan(ndarray a, ndarray out=None):
+cdef ndarray scan(
```

Comment:
Could you rename this function to make the deference to `_cum_core`? (How about `scan_1d` for example?)